src/App.tsx

```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isSetReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
+
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return (!isSetReversed) ? visibleGoods : visibleGoods.reverse();
+}
+
 export const App: React.FC = () => {
+  const {
+    ALPHABET,
+    LENGTH,
+    NONE,
+  } = SortType;
+
+  const [sortType, setSortType] = useState(NONE);
+  const [isSetReversed, setReverse] = useState(false);
+
+  const goods = getReorderedGoods(goodsFromServer, sortType, isSetReversed);
+  const visibilityReset = isSetReversed || sortType !== NONE;
```

Comment:
It stores a boolean value, so should be called according to these rules
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig
![image](https://user-images.githubusercontent.com/96409563/222951094-1ed0b879-4ccb-4a3f-8f3e-5c414929440d.png)